src/types/bytes.rs

```diff
@@ -80,10 +113,23 @@ impl PyBytes {
     /// `std::slice::from_raw_parts`, this is
     /// unsafe](https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html#safety).
     pub unsafe fn from_ptr(py: Python<'_>, ptr: *const u8, len: usize) -> &PyBytes {
```

Comment:
Since this returns a gil-ref, should we deprecate it as well, in favor of `bound_from_ptr()`?

Reply:
Yes, my mistake to miss this; that's the downside of the commits I've been cherry-picking from #3606 - they were mostly trying to get a complete compile with a good range of the types rather than a complete implementation 🙈 

I'll add here 👍